cmd/validator/flags/flags.go

```diff
@@ -364,6 +364,13 @@ var (
 		Usage: "Sets gas limit for the builder to use for constructing a payload for all the validators",
 		Value: fmt.Sprint(params.BeaconConfig().DefaultBuilderGasLimit),
 	}
+
+	// ValidatorRegistrationBatchSizeFlag sets the maximum size for one batch of validator registrations. Use a non-positive value to disable batching.
+	ValidatorRegistrationBatchSizeFlag = &cli.IntFlag{
+		Name:  "validator-registration-batch-size",
```

Comment:
Is it the case that if a BN gets many registrations from different VCs then the problem of large requests timing out won't happen because each such request is sent individually? I ask cause it's not clear to me where the natural place for this gating is.  Either the validator client or the beacon that is actually sending the registrations. But if the BN serving several VCs already acts as chunking then this should be fine.

Reply:
Just for information, what I saw is not relays timeout, but relays error, with something as "too large payload". (But it does not change the heart of the issue.)

> Is it the case that if a BN gets many registrations from different VCs then the problem of large requests timing out won't happen because each such request is sent individually? 

Yes. (Unless one of the VC manages itself a large number of keys, and sends only one big query to the corresponding BN, which is, before this PR, the case with Prysm.)

> I ask cause it's not clear to me where the natural place for this gating is. 

I asked myself the same question before I started to worked on this issue. I did not find clear pros/cons to batch requests in VC or in BN. So the choice to do it in the VC was mainly dictated by the fact that Lighthouse already does it in VC. A node operator using multiple implementations will be happy to set the same flag, in the same client (here, the VC).